internal/pkg/deploy/cloudformation/stack/lb_web_svc.go

```diff
@@ -62,64 +60,62 @@ type LoadBalancedWebService struct {
 	parser loadBalancedWebSvcReadParser
 }
 
-// LoadBalancedWebServiceOption represents an option to apply to a LoadBalancedWebService.
-type LoadBalancedWebServiceOption func(s *LoadBalancedWebService)
-
-// WithHTTPS enables HTTPS for a LoadBalancedWebService. It creates an HTTPS listener and assumes that the environment
-// the service is being deployed into has an HTTPS configured listener.
-func WithHTTPS() func(s *LoadBalancedWebService) {
-	return func(s *LoadBalancedWebService) {
-		s.dnsDelegationEnabled = true
-		s.httpsEnabled = true
-	}
-}
-
-// WithNLB enables Network Load Balancer in a LoadBalancedWebService.
-func WithNLB(cidrBlocks []string) func(s *LoadBalancedWebService) {
-	return func(s *LoadBalancedWebService) {
-		s.publicSubnetCIDRBlocks = cidrBlocks
-	}
-}
-
-// WithDNSDelegation enables DNS delegation for a LoadBalancedWebService.
-func WithDNSDelegation(app deploy.AppInformation) func(s *LoadBalancedWebService) {
-	return func(s *LoadBalancedWebService) {
-		s.dnsDelegationEnabled = true
-		s.appInfo = app
-	}
+// NewLoadBalancedWebServiceOpts contains fields to configure LoadBalancedWebService.
+type NewLoadBalancedWebServiceOpts struct {
```

Comment:
nit: nice, i like this too - way simpler, just a nit
```suggestion
type LoadBalancedWebServiceOpts struct {
```